subsys/net/lib/app/init.c

```diff
@@ -252,3 +237,36 @@ int net_sample_app_init(const char *app_info, u32_t flags, s32_t timeout)
 
 	return 0;
 }
+
+#if defined(CONFIG_NET_APP_AUTO_INIT)
+static int init_net_app(struct device *device)
+{
+	u32_t flags = 0;
+	int ret;
+
+	ARG_UNUSED(device);
+
+	if (IS_ENABLED(CONFIG_NET_APP_NEED_IPV6)) {
+		flags |= NET_APP_NEED_IPV6;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_APP_NEED_IPV6_ROUTER)) {
+		flags |= NET_APP_NEED_ROUTER;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_APP_NEED_IPV4)) {
+		flags |= NET_APP_NEED_IPV4;
+	}
+
+	/* Initialize the application automatically if needed */
+	ret = net_app_init("Initializing network", flags,
+			   K_SECONDS(CONFIG_NET_APP_INIT_TIMEOUT));
+	if (ret < 0) {
+		NET_ERR("Network initialization failed (%d)", ret);
```

Comment:
I'd suggest to make this printk() and k_panic(), to make sure this cannot be missed, regardless of logging options.